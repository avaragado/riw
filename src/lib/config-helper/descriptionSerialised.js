// @flow

import stringify from 'json-stable-stringify';

import type {
    Description,
    SerialisedDescription,
} from '../../types';
import { sDescriptionDefault } from '../riw';

export default (desc: ?Description = sDescriptionDefault): SerialisedDescription => (
    typeof desc === 'string'
        ? desc
        : stringify(desc)
);
