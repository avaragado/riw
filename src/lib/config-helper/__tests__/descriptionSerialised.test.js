// @flow

import type { Description } from '../../../types';
import descriptionSerialised from '../descriptionSerialised';

type Fixture = {
    name: string,
    in?: Description,
};

const fixtures: Fixture[] = [
    {
        name: '01 - undefined becomes default description',
    },

    {
        name: '02 - string remains as-is',
        in: 'foobar baz',
    },

    {
        name: '03 - object is stringified with sorted keys',
        in: {
            z: 1,
            a: 2,
            nested: {
                z: 3,
                a: 4,
                m: 5,
            },
            m: 6,
        },
    },

    {
        name: '04 - object is stringified with sorted keys',
        in: {
            a: 2,
            m: 6,
            nested: {
                a: 4,
                m: 5,
                z: 3,
            },
            z: 1,
        },
    },
];

describe('lib/config-helper/descriptionSerialised', () => {
    fixtures.forEach((fixture) => {
        it(fixture.name, () => {
            const received = descriptionSerialised(fixture.in);

            expect(received).toMatchSnapshot();
        });
    });
});
