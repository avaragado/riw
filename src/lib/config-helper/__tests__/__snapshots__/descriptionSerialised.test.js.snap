// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/config-helper/descriptionSerialised 01 - undefined becomes default description 1`] = `"_"`;

exports[`lib/config-helper/descriptionSerialised 02 - string remains as-is 1`] = `"foobar baz"`;

exports[`lib/config-helper/descriptionSerialised 03 - object is stringified with sorted keys 1`] = `"{\\"a\\":2,\\"m\\":6,\\"nested\\":{\\"a\\":4,\\"m\\":5,\\"z\\":3},\\"z\\":1}"`;

exports[`lib/config-helper/descriptionSerialised 04 - object is stringified with sorted keys 1`] = `"{\\"a\\":2,\\"m\\":6,\\"nested\\":{\\"a\\":4,\\"m\\":5,\\"z\\":3},\\"z\\":1}"`;
